internal/hub/conf.go

```diff
@@ -0,0 +1,208 @@
+	BundleLabelKey EnvVarKey = iota
+	ClientIDKey
+	ClientSecretKey
+	OfflineKey
+	PDPIDKey
+	WorkspaceSecretKey
+)
+
+var envVars = map[EnvVarKey][]string{
+	BundleLabelKey:     {"CERBOS_HUB_BUNDLE", "CERBOS_CLOUD_BUNDLE"},
+	ClientIDKey:        {"CERBOS_HUB_CLIENT_ID", "CERBOS_CLOUD_CLIENT_ID"},
+	ClientSecretKey:    {"CERBOS_HUB_CLIENT_SECRET", "CERBOS_CLOUD_CLIENT_SECRET"},
+	OfflineKey:         {"CERBOS_HUB_OFFLINE", "CERBOS_CLOUD_OFFLINE"},
+	PDPIDKey:           {"CERBOS_HUB_PDP_ID", "CERBOS_PDP_ID"},
+	WorkspaceSecretKey: {"CERBOS_HUB_WORKSPACE_SECRET", "CERBOS_CLOUD_SECRET_KEY"},
+}
+
+func GetEnv(key EnvVarKey) string {
+	varNames, ok := envVars[key]
+	if !ok {
+		return ""
+	}
+
+	for _, v := range varNames {
+		val, ok := os.LookupEnv(v)
+		if ok {
+			return val
+		}
+	}
+
+	return ""
+}
+
+const (
+	confKey                  = "hub"
+	defaultAPIEndpoint       = "https://api.cerbos.cloud"
+	defaultBootstrapHost     = "https://cdn.cerbos.cloud"
+	defaultHeartbeatInterval = 180 * time.Second
+	defaultMaxRetryWait      = 120 * time.Second
+	defaultMinRetryWait      = 1 * time.Second
+	defaultNumRetries        = 5
+	minHeartbeatInterval     = 30 * time.Second
+)
+
+type Conf struct {
+	// Credentials holds Cerbos Hub client credentials.
+	Credentials CredentialsConf `yaml:"credentials"`
+	// Connection holds advanced connection settings for Cerbos Hub.
+	Connection ConnectionConf `yaml:"connection"`
+}
+
+func (conf *Conf) Key() string {
+	return confKey
+}
+
+func (conf *Conf) SetDefaults() {
+	conf.Credentials = CredentialsConf{
+		ClientID:        GetEnv(ClientIDKey),
+		ClientSecret:    GetEnv(ClientSecretKey),
+		PDPID:           GetEnv(PDPIDKey),
+		WorkspaceSecret: GetEnv(WorkspaceSecretKey),
+	}
+}
+
+func (conf *Conf) Validate() (outErr error) {
+	_ = conf.Connection.Validate()
+	return conf.Credentials.Validate()
+}
+
+// CredentialsConf holds credentials for accessing Cerbos Hub.
+type CredentialsConf struct {
+	// PDPID is the unique identifier for this Cerbos instance. Defaults to the value of the CERBOS_HUB_PDP_ID environment variable.
+	PDPID string `yaml:"pdpID" conf:",example=crb-004"`
+	// ClientID of the Cerbos Hub credential. Defaults to the value of the CERBOS_HUB_CLIENT_ID environment variable.
+	ClientID string `yaml:"clientID" conf:",example=92B0K05B6HOF"`
+	// ClientSecret of the Cerbos Hub credential. Defaults to the value of the CERBOS_HUB_CLIENT_SECRET environment variable.
+	ClientSecret string `yaml:"clientSecret" conf:",example=${CERBOS_HUB_CLIENT_SECRET}"`
+	// WorkspaceSecret used to decrypt the bundles. Defaults to the value of the CERBOS_HUB_WORKSPACE_SECRET environment variable.
+	WorkspaceSecret string `yaml:"workspaceSecret" conf:",example=${CERBOS_HUB_WORKSPACE_SECRET}"`
+	// Deprecated: Use PDPID
+	InstanceID string `yaml:"instanceID" conf:",ignore"`
+	// Deprecated: Use WorkspaceSecret
+	SecretKey string `yaml:"secretKey" conf:",ignore"`
+}
+
+func (cc *CredentialsConf) Validate() (outErr error) {
```

Comment:
Also here?

Reply:
Same as above, this swap around has to be done at the end of config loading.